invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalConsumableUsageInArrear.java

```diff
@@ -227,18 +225,18 @@ protected UsageInArrearAggregate getToBeBilledUsageDetails(final LocalDate start
             // We generate an entry if we consumed anything on this tier or if this is the first tier to also support $0 Usage item
             if (hasPreviousUsage) {
                 // FIXME-1469 change to correct BigDecimal implementation
-                final Long previousUsageQuantity = tierNum <= lastPreviousUsageTier ? previousUsage.get(tierNum - 1).getQuantity().longValue() : 0;
+                final BigDecimal previousUsageQuantity = tierNum <= lastPreviousUsageTier ? previousUsage.get(tierNum - 1).getQuantity() : BigDecimal.ZERO;
```

Comment:
The current code is converting the  `tieredBlock#Max` (double) into a `long`, later requiring the conversion `previousUsageQuantity#longValue`. I think the options are to
1. Also update our [catalog api](https://github.com/killbill/killbill-api/blob/master/src/main/java/org/killbill/billing/catalog/api/TieredBlock.java#L24) to specify a `BigDecimal`, OR
2. Make a conversion   `tieredBlock#Max` (double) into a `BigDecimal` avoiding the conversion you need later `previousUsageQuantity#longValue`